ms2deepscore/data_generators.py

```diff
@@ -226,6 +197,8 @@ def __getitem__(self, batch_index: int):
         if self.settings['use_fixed_set'] and batch_index in self.fixed_set:
             return self.fixed_set[batch_index]
         if self.settings['use_fixed_set'] and batch_index == 0:
+            # todo This sets the global random seed to 42. Wouldn't this cause an issue, since this random seed will
```

Comment:
@florian-huber I am not fully sure but I feel like setting the random seed is a global setting. Therefore any other datagenerator will also use this random seed. 
This would mean that once you use a fixed set datagenerator (which we do for the validation test set). After it has reached this point all other datagenerators will also become fixed, since they will also use seed 42. 

I am not fully sure it this is indeed a global setting and if there is something preventing this, but it certainly felt risky.

Reply:
Even if we do not set it in our function there is also a risk that any of the other modules do actually set a random seed. Which would affect how the method functions. 
After googling I found a blog post about this issue including some solutions: 
https://builtin.com/data-science/numpy-random-seed

Reply:
hmm, but when calling the np.random function multiple times it will still create different numbers every time. So it might not be an issue in that case. (I have to think about it a bit more :) ).

Reply:
That is indeed a bit tricky. 
The cleanest way would maybe be to initiate a random number sequence (as in the blog post you mentioned), but since we run a lot of different randomized functions (`choice`, `random`, `randint`) that is maybe not that straightforward.

You can make an issue on this. But since the unit test on this shows that the desired behavior is achieved with the current code I wouldn't change it in this PR.

Reply:
Actually, I now added `random_seed` to the settings. Now:
- `fixed_set = True` and `random_seed = ...` --> same fixed order for every generator run and same order for each batch
- only `fixed_set = True`  --> same order for each batch (but different for each new generator initialization and run)